File: src/003/java/003.java
import java.math.*;
import java.util.stream.*;

public class PrimeFactors {
  public static void main(String... args) {

    long target = 600851475143l;

    Long l = LongStream
      .range(1, (long) Math.floor(Math.sqrt(target)))
      .filter(i -> BigInteger.valueOf(i).isProbablePrime(5))
      .filter(i -> target % i == 0)
      .max()
      .orElse(0);

    System.out.println(l);

  }
}

File: src/003/python/003.py
from math import floor, sqrt

t = 600851475143

def isprime(n):
    for i in range(2, n+1//2):
        if n % i == 0:
            return False
    return True

for d in reversed(range(2, floor(sqrt(t)))):
    if t % d == 0 and isprime(d):
        print(d)
        break
